static/js/server_events.js

```diff
@@ -236,6 +236,21 @@ function get_events(options) {
                                      save_narrow: true,
                                      save_compose: true});
                 }
+                // If this browser window were incorrectly
+                // re-initialized from a browser cache, it will have a
+                // stale `last_event_id` in page_params, which will
+                // result in the REQUEST_PRUNED_EVENTS error from the server.
+                //
+                // In this case, we should destroy our event queue and
+                // reload the browser window.
+                if (xhr.status === 400 &&
+                    JSON.parse(xhr.responseText).code === 'REQUESTED_PRUNED_EVENTS') {
+                    exports.cleanup_event_queue();
+                    reload.initiate({immediate: true,
```

Comment:
Probably low importance, but does the asynchronous `cleanup_event_queue` request actually make it to the server before the browser reloads? For that matter, the same question applies to both existing callers of `cleanup_event_queue`.

Reply:
Empirically, it does; we tested this pretty extensively when we introduced `cleanup_event_queue` in the first place.  I think because all we need is for the browser to start the request.  I could imagine some more complex thing involving local storage or waiting for the request to succeed, but the additional complexity does create significant risk of bugs.